startup/gui/usd.py

```diff
@@ -45,3 +53,48 @@
 # `texture:format == automatic` isn't well supported at present, so default
 # user-created lights to `latlong`.
 Gaffer.Metadata.registerValue( GafferUSD.USDLight, "parameters.texture:format", "userDefault", "latlong" )
+
+
+def __kindSelectionModifier( targetKind, scene, pathString ) :
+	path = pathString.split( "/" )[1:]
+
+	kind = None
+	while len( path ) > 0 :
+		attributes = scene.attributes( IECore.InternedStringVectorData( path ) )
+		kind = attributes.get( "usd:kind", None )
+
+		if kind is not None and Kind.Registry.IsA( kind.value, targetKind ) :
+			break
+		path.pop()
+
+	return "/" + "/".join( path )
```

Comment:
Likewise, I think here we might just be able to return `path` now?

Reply:
Same as above, fixed in https://github.com/GafferHQ/gaffer/pull/5720/commits/cde068adae72c742305aab275801a2973a37e267